crates/stackable-operator/CHANGELOG.md

```diff
@@ -12,6 +12,20 @@ All notable changes to this project will be documented in this file.
 
 - BREAKING: The `CustomResourceExt` trait is now re-exported from the `stackable-shared` crate. The
   trait functions use the same parameters but return a different error type ([#883]).
+- BREAKING: `KeyValuePairs<V>` (and `Annotations`/`Labels`) is now an alias for `BTreeMap<Key, V>` ([#889]).
+  - Generally, this means that the API now behaves like a map rather than a set. For example, duplicate keys are no longer allowed (as was already documented before).
+  - Some `KeyValuePairs` methods have been renamed for certain use cases:
+    - `KeyValuePairs::insert(&mut self, kvp)`: use `::extend(&mut self, [kvp])` instead.
+    - `KeyValuePairs::try_from`: you may need to use `::try_from_iter` instead.
+    - `KeyValuePairs::contains_key`: unvalidated keys will need to use `::contains_str_key` instead.
+    - `Into<BTreeMap<String, String>>`: use `::to_unvalidated` instead.
```

Comment:
note: I can see why this name is chosen, but can maybe be a little misleading. The source is obviously valid, the result is also valid (if not mutated).

Let's discuss some alternate names, like `to_btree_map` for example.